Third epic first story create migrations for tables: english_lvl, students, feedbacks

voldan opened the pull request:
migrations for 3 new tables: english_lvl, students, feedbacks

remade datademo command - now the filling of tables is performed by one command.

voldan commented:
3 issues to fix:
- Class name "m170901_082020_create_english_lvl_table" is not in camel caps format
- Class name "m170901_083320_create_students_table" is not in camel caps format
- Class name "m170901_104716_create_feedbacks_table" is not in camel caps format

the class name is generated by Yii framework, I can't rename those files. If I will do it - migrations will not work properly.

boonya commented on protected/migrations/m170901_104716_create_feedbacks_table.php:
```diff
@@ -0,0 +1,32 @@
+                    'teacher_score' => 'integer NOT NULL',
+                    'teacher_summery' => 'varchar(250) NOT NULL',
+                    'expert_score' => 'integer NOT NULL',
+                    'expert_summery' => 'varchar(250) NOT NULL',
```
typo. Has to be an `expert_summary`

boonya commented on protected/migrations/m170901_104716_create_feedbacks_table.php:
```diff
@@ -0,0 +1,32 @@
+                    'id' => 'pk',
+                    'student_id' => 'integer NOT NULL',
+                    'teacher_score' => 'integer NOT NULL',
+                    'teacher_summery' => 'varchar(250) NOT NULL',
```
As below

voldan commented:
I added a migration in order to fix typos in 'feedbacks' table, and refactored code in DemoDataCommand.php file.
Codeclimate issue: Class name "m170905_185914_feedback_table_typo_fix" is not in camel caps format - is not a fixable issue.

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
-        foreach ($locations as $location) {
-            $command->insert('locations', $location);
-        }
+        return $demoData = require_once Yii::app()->basePath . '/data/demo/' . $fileName;
```
`return require_once Yii::app()->basePath . "/data/demo/{$fileName}.php";`

boonya replied:
```
$filePath = Yii::app()->basePath . "/data/demo/{$fileName}.php";
if (!file_exists($filePath)) {
    throw new RuntimeException("File {$filePath} does not exist!");
}
return require_once $filePath;
```
better

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
+    private function fillOutTable($tableName)
     {
-        $users = require_once Yii::app()->basePath . '/data/demo/users.php';
+        $fileName = $tableName . '.php';
```
It is redundant line if you'll implement method as I said above.

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
     }
 
-    public function actionFillOutTableGroups()
+    public function actionFillOutTableUsers($tableName)
```
This method contains a word `users` in his name. Why did you pass a variable into it? It is pretty obvious that this method has to work with users data!

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
     }
 
-    public function actionFillOutTableUserGroups()
+    public function actionFillOutTableDirections($tableName)
```
same

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
     }
 
-    public function actionFillOutTableUserRoles()
+    public function actionFillOutTableGroups($tableName)
```
same

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
     }
 
-    public function actionFillOutTableExperts()
+    public function actionFillOutTableUserGroups($tableName)
```
same

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
     }
 
-    public function actionFillOutTableStudents()
+    public function actionFillOutTableUserRoles($tableName)
```
same

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
+        $this->fillOutTable($tableName);
+    }
+
+    public function actionFillOutTableExperts($tableName)
```
same

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,95 +2,70 @@
+        $this->fillOutTable($tableName);
+    }
+
+    public function actionFillOutTableStudents($tableName)
```
same

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,65 +2,70 @@
+    private function getDemoData($fileName)
     {
-        $locations = require_once Yii::app()->basePath . '/data/demo/locations.php';
+        return require_once Yii::app()->basePath . '/data/demo/' . $fileName;
```
Why not `return require_once Yii::app()->basePath . "/data/demo/{$fileName}.php";`?
But:
```
$filePath = Yii::app()->basePath . "/data/demo/{$fileName}.php";
if (!file_exists($filePath)) {
    throw new RuntimeException("File {$filePath} does not exist!");
}
return require_once $filePath;
```
much better

boonya commented on protected/commands/DemoDataCommand.php:
```diff
@@ -2,65 +2,70 @@
+
+    private function fillOutTable($tableName)
+    {
+        $fileName = $tableName . '.php';
```
It is redundant line if you'll implement method as I said above.

boonya commented on protected/migrations/m170901_082020_create_english_lvl_table.php:
```diff
@@ -0,0 +1,30 @@
+            ['id' => 1, 'name' => 'elementary'],
+            ['id' => 2, 'name' => 'pre-intermediate'],
+            ['id' => 3, 'name' => 'intermediate'],
+            ['id' => 4, 'name' => 'upper- intermediate'],
```
_MINOR:_ typo. I believe it has to be an `upper-intermediate`
